src/aggregate.d

```diff
@@ -270,13 +276,15 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
@MartinNowak - it's bad enough that recursive calls to `determineFields` calls `fields.setDim(0)` invalidating any work that is being done up the call stack.  But that's recursive forward referencing for you.

The return is is precisely because all fields were determined whilst analysing the current member.  Analysing any more members may result in fields being added twice into the fields array.